src/main/java/org/jenkinsci/remoting/protocol/impl/BIONetworkLayer.java

```diff
@@ -273,6 +287,13 @@ public void run() {
                 } finally {
                     release(buffer);
                 }
+            } catch (Throwable e) {
```

Comment:
🐛 needs rethrow after logging. Otherwise propagation of serious issues like OOM may be suppressed